src/main/kotlin/graphics/scenery/attribute/geometry/Geometry.kt

```diff
@@ -41,38 +45,22 @@ interface Geometry : Serializable {
      * STL's facet storage format into account.
      */
     fun recalculateNormals() {
-        val vertexBufferView = vertices.asReadOnlyBuffer()
-        var i = 0
-        val normals = ArrayList<Float>()
+        var pVtx = vertices.adr.toPtr<Vector3f>()
+        var pNorm = normals.adr.toPtr<Vector3f>()
 
-        while (i < vertexBufferView.limit() - 1) {
-            val v1 = Vector3f(vertexBufferView[i], vertexBufferView[i + 1], vertexBufferView[i + 2])
-            i += 3
-
-            val v2 = Vector3f(vertexBufferView[i], vertexBufferView[i + 1], vertexBufferView[i + 2])
-            i += 3
-
-            val v3 = Vector3f(vertexBufferView[i], vertexBufferView[i + 1], vertexBufferView[i + 2])
-            i += 3
+        for (i in vertices.indices step 3) {
+            val v1 = pVtx++[0]
+            val v2 = pVtx++[0]
+            val v3 = pVtx++[0]
```

Comment:
This is super concise, but I find it harder to understand. Maybe a middle ground could be:
```kotlin
fun makeVector(array, i) { return Vector3f(array[i], array[i + 1], array[i + 2]) }

val v1 = makeVector(vertexBufferView, i += 3)
val v2 = makeVector(vertexBufferView, i += 3)
val v3 = makeVector(vertexBufferView, i += 3)
```
or some such? Or even attach a `toVector(i)` method to the `FloatBuffer` class here?

It is fun to use overloaded operators, but the pointer thing with `[0]` is strange.

Reply:
it's essentially `*pVtx++`

Another alternative:

```
        for (i in vertices.indices step 3) {
            val v1 = pVtx[i]
            val v2 = pVtx[i + 1]
            val v3 = pVtx[i + 2]
```

Reply:
Nice, I like that much more!